astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
```

Comment:
By assuming `PyUnitListProxy` here instead of `PyObject` (see [`richcmpfunc`](https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_richcompare)) we might get segfaults if you compare them to non `PyUnitListProxy`. (could you add a test case for a different class?)